### PR TITLE
Bug fix/add employee

### PR DESCRIPTION
**Pull request by fawadmehmood1**
# Description

Please include a summary of the changes and the related issue.
- Add Employee Form bugs
- Middleware Temporary fix

## Type of change

Please select type of issue addressed,

- [x] Bug fix (MLT-69, MLT-78, MLT-80, MLT-82, MLT-83, MLT-84, MLT-85)

# Dependences / Libraries Installed

Please list the packages and libraries installed (if any)

- [x] ( @radix-ui/react-scroll-area - Added by shadcn) 

# How Has This Been Tested By The Developer?
- Verified if all the bugs are resolved

# Checklist:

- [x] I have performed a self review of my code.

- [x] I have removed consoles, debuggers and unnecessary code blocks.

- [x] My code follows the style guidelines of this project

- [x] My changes generate no new warnings or errors

- [x] I have commented my code, particularly in hard-to-understand areas

- [ ] New and existing unit tests pass locally with my changes

- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I understand that I am accountable for any code added in this PR, and any negative impact will effect my performance evaluation

### CODE REVIEW

File: components/employees/add-employee-form/index.tsx [hunk 1]
```diff
@@ -77,11 +101,8 @@ const AddEmployee: FC<AddEmployeeProps> = ({ setOpenEmployeeForm }) => {
 
   return (
     <Form {...form}>
-      <form
-        onSubmit={form.handleSubmit(onSubmit)}
-        className="h-full max-h-[80vh] overflow-y-auto px-6 pb-0 pt-[2.375rem] md:px-11"
-      >
-        <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:gap-6">
+      <form onSubmit={form.handleSubmit(onSubmit)} className=" ">
```
Comment (imrankhan-stixor) [inline components/employees/add-employee-form/index.tsx]: remove empty className

File: components/employees/add-employee-form/index.tsx [hunk 2]
```diff
@@ -376,16 +414,24 @@ const AddEmployee: FC<AddEmployeeProps> = ({ setOpenEmployeeForm }) => {
...
-            name={"contractEnd"}
+            name={"contractExpiryDate"}
             render={({ field }) => (
               <FormItem>
                 <FormLabel>{contractExpiryDate?.label}</FormLabel>
 
                 <FormControl>
                   <DatePicker
-                    onSelect={field.onChange}
+                    onSelect={(date) => {
```
Comment (imrankhan-stixor) [inline components/employees/add-employee-form/index.tsx]: this code is used thrice so make it a function and pass date as prop

File: components/employees/add-employee-form/index.tsx [hunk 3]
```diff
@@ -376,16 +414,24 @@ const AddEmployee: FC<AddEmployeeProps> = ({ setOpenEmployeeForm }) => {
...
+                      field.onChange(date);
+                      setContractEndDate(date);
+                    }}
                     placeholder={contractExpiryDate?.placeholder}
                     value={field?.value}
+                    disabled={(date: Date) =>
+                      date < new Date("1900-01-01") ||
+                      (hiringDate ? date <= hiringDate : false) ||
+                      (probationEndDate ? date < probationEndDate : false)
+                    }
```
Comment (imrankhan-stixor) [inline components/employees/add-employee-form/index.tsx:449]: this is also used thrice so make it a function and use dates as props

**Review by imrankhan-stixor — CHANGES_REQUESTED** [review]

File: components/employees/add-employee-form/index.tsx (hunk 3 above)
Comment (fawadmehmood1) [reply components/employees/add-employee-form/index.tsx:449]: @imrankhan-stixor, `Disabled` covers different conditions checked for each data picker. If we create a function, we'd need to add switch/if checks to determine which conditions to run. Thus, there's no harm in handling them separately, as we aren't reusing the same checks elsewhere.

Comment (imrankhan-stixor) [reply components/employees/add-employee-form/index.tsx:449]: you're to make a function like this:

const disableTesting=(data:Date, initialTestDate:Date, secondaryTestDate:Date)=>{
return  date < new Date("1900-01-01") ||
                      (initialTestDate ? date <= initialTestDate : false) ||
                      (secondaryTestDate ? date < secondaryTestDate : false)
}

**Review by imrankhan-stixor — CHANGES_REQUESTED** [review]

File: components/ui/scroll-area.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    ref={ref}
+    orientation={orientation}
+    className={cn(
+      "flex touch-none select-none transition-colors",
+      orientation === "vertical" &&
+        "h-full w-2.5 border-l border-l-transparent p-[1px]",
+      orientation === "horizontal" &&
+        "h-2.5 flex-col border-t border-t-transparent p-[1px]",
+      className,
+    )}
```
Comment (hassantahir767) [inline components/ui/scroll-area.tsx:22]: define scrollbar above and then use below to avoid warning.

**Review by hassantahir767 — CHANGES_REQUESTED** [review]

File: components/employees/add-employee-form/index.tsx (hunk 3 above)
Comment (fawadmehmood1) [reply components/employees/add-employee-form/index.tsx:449]: conditions are not same

**Review by imrankhan-stixor — APPROVED** [review]

**Review by hassantahir767 — APPROVED** [review]

**Pull request merged**